src/perf.c

```diff
@@ -69,7 +69,7 @@ int perfopen(pid_t pid, const PerfOption *opt, int cpu, bool extras) {
   attr.freq = opt->freq;
 
   // Breakpoint
-  if (opt->type & PERF_TYPE_BREAKPOINT) {
+  if (opt->type == PERF_TYPE_BREAKPOINT) {
```

Comment:
Ah, good catch.  This was my mistake.  Many of the configurations for these interfaces are bitmasks, but this is not the case with `perf_event_open()`'s type field of its input attr.

For a similar mistake, check out https://github.com/DataDog/ddprof/blob/main/src/ddprof_worker.cc#L161.  There we correctly use `==`, but checked the enum against the `type` field rather than the `config` field.  The consequence of the failed check are rather limited there as well.